pkg/restapi/kms/operation/operations.go

```diff
@@ -217,7 +217,8 @@ func writeErrorResponse(rw http.ResponseWriter, status int, msg string) {
 
 func keystoreLocation(hostURL, keystoreID string) string {
 	// {hostURL}/kms/keystores/{keystoreID}
-	return fmt.Sprintf("%s/%s/%s", hostURL, keystoreEndpoint, url.PathEscape(keystoreID))
+	return fmt.Sprintf("%s%s", hostURL,
+		strings.ReplaceAll(keystoreEndpoint, "{keystoreID}", url.PathEscape(keystoreID)))
```

Comment:
Could you just PathEscape the entire URL?

Reply:
KMS key IDs are base64URL encoded, I recommend `keytstoreID` be of the same format so that we don't have to deal with escaping for URL values.

Reply:
I was considering to use a schema like "urn:uuid:85149342-7f26-4dc1-a77a-345f4a1102d5" as for [StructuredDocument](https://identity.foundation/secure-data-store/#structureddocument), but now I see that it's much more sense to use the one that [SDS vault](https://identity.foundation/secure-data-store/#example-8-successful-data-vault-creation-response) uses. And it's the one that you are talking about. So, I will switch to it.

P.S. I am referring to SDS because spec for WebKMS is not useful currently and, for me, those two systems look quite similar.

Reply:
@Baha-sk is the algorithm that you are using for keys [this one](https://github.com/hyperledger/aries-framework-go/blob/bcf1011f3c7071c723681eed58ea3352eaa212ab/pkg/kms/localkms/localkms_writer.go#L86)

Reply:
updated using https://github.com/rs/xid lib

Reply:
@aholovko Note that I believe there's a mistake in the SDS spec - it shows a UUID in the StructuredDocument example but according to the text in the table above it, it should be a Base58-encoded 128-bit random value

Reply:
> updated using https://github.com/rs/xid lib

@aholovko fyi, the code snippet you pasted in the previous post will change, I've updated the `kid` to be jwk thumbprint (for asymmetric keys only) base64URL encoded. Symmetric keys will keep the KID in the link you pasted above (except the new length will be 50 now). Not sure if rs/xid project you mentioned is more advantageous.